src/core/water.cpp

```diff
@@ -422,3 +432,85 @@ void Water::ReturnPadding(RotationMatrix max_rotation, float in_plane_rotation,
         }
     }
 }
+
+namespace {
+
+void set_matrix_identity(float* matrix, int n) {
+    std::memset(matrix, 0, n * n * sizeof(float));
+    for ( int i = 0; i < n; i++ ) {
+        matrix[i * n + i] = 1.0f;
+    }
+}
+
+} // namespace
+
+void Water::ComputeTransformedPositions(RotationMatrix rotation) {
```

Comment:
Transform the water positions into the global coordinate system. This is done using a 4x4 matrix computing an affine transform of the position, which is composed of a rotation + translation.

We construct the matrix, then use a large BLAS call to apply the matrix to the position vectors.